dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -195,4 +195,19 @@ default void checkOrganization(Organization organization) {
         }
 
     }
+
+    /**
+     * Only passes if entry is published or if user has correct credentials
+     * @param user Optional user
+     * @param entry Entry to check
+     */
+    default void optionalUserCheckEntry(Optional<User> user, Entry entry) {
+        if (!entry.getIsPublished()) {
+            if (user.isEmpty()) {
+                throw new CustomWebApplicationException("Entry not found", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Shouldn't this be a 404 instead of a 400?

Hmm, just noticed that `checkEntry` throws 400 as well. This might be opening a can of worms.